### ISSUE TITLE
`useAsyncData` clears out nested route data before leaving

### ISSUE THREAD

**Issue by mvrlin** [nuxt/nuxt#13814]
### Environment

------------------------------
- Operating System: `Darwin`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-rc.1`
- Package Manager:  `pnpm@6.32.8`
- Builder:          `vite`
- User Config:      `build`, `buildModules`, `components`, `css`, `srcDir`, `typescript`, `vite`, `server`
- Runtime Modules:  `-`
- Build Modules:    `@intlify/nuxt3@0.1.10`, `@pinia/nuxt@0.1.8`, `nuxt-windicss@2.3.1`
------------------------------

### Reproduction

https://stackblitz.com/edit/github-ypbf5b?file=app.vue
To reproduce the issue, load the website with the url `/users/foo` and click **Go Home**

### Describe the bug

`useAsyncData` clears out nested route data before leaving. If the next route is still loading, it produces a blank page, which isn't great 😢 Maybe it is due to provide/inject.

### Additional context

https://user-images.githubusercontent.com/24970784/164979549-cdb21c9c-f791-4a6a-9c7d-a1084f2c78e8.mov

**Comment by champ7champ** [comment]
Such an important problem has not been solved in a year, and there is not even a temporary fix. Probably have to say goodbye to nuxt

**Comment by danielroe** [comment]
I'm sorry you feel that way. As you see, it is an [upstream bug](https://github.com/vuejs/core/issues/5513) which we have [created a PR](https://github.com/vuejs/core/pull/6736) to fix, but it depends on Vue itself merging it.

**Comment by champ7champ** [comment]
@danielroe How long do you think it will take to solve this problem? Week, month, year?

**Comment by manniL** [comment]
@champ7champ Whenever the linked PR will be merged, which is not in our hands.

**Comment by mvrlin** [comment]
> @champ7champ Whenever the linked PR will be merged, which is not in our hands. You could probably monkey-patch Vue in the meantime as "workaround" if really necessary.

The patch doesn't work. Maybe the problem is not only with `Suspense`, but `KeepAlive` also.

**Comment by danielroe** [comment]
Once that PR is merged, we then have to update Nuxt's implementation to _use_ the prop it adds.

**Comment by manniL** [comment]
> Once that PR is merged, we then have to update Nuxt's implementation to _use_ the prop it adds.

Whoops, good point 🙈

**Comment by champ7champ** [comment]
> Once that PR is merged, we then have to update Nuxt's implementation to _use_ the prop it adds.

OMG!!! How much longer to wait? A new framework will be released sooner than we will wait for this fix ((( 
How to influence this???

**Comment by danielroe** [comment]
The implementation will be pretty much instant on the Nuxt end 😆 The best thing you can do is not comment on this issue as it pings everyone who is subscribed.

**Comment by mvrlin** [comment]
> The implementation will be pretty much instant on the Nuxt end 😆 The best thing you can do is not comment on this issue as it pings everyone who is subscribed.

I patched the Vue with `npx patch-package` with the PR you provided. Can you give a little instruction what to change in the Nuxt package to make it work? I tried to change the NuxtPage by adding suspensible prop, but it doesn't seem to work.

**Comment by champ7champ** [comment]
Activity is falling, the problem is moving further and further away, or is something being solved there?

**Comment by Luffyyy** [comment]
> msedge_eW8BVcEJpZ.mp4
> 
> Not sure if this is the same issue, but I seem to experience something similar.
> 
> Reproduction; https://stackblitz.com/edit/github-f4gpfk-m2cxo3?file=app.vue

@danielroe This still happens in the latest build, is this intended behavior? where you leave the route and the child gets detached first while the parent isn't? It just looks plain weird.

**Comment by danielroe** [comment]
It's definitely not intended behaviour, but I think it's also different from the original issue here. I've opened a new issue to keep track of it.
